### PR TITLE
chore(docs): add headless auth usage page

### PR DESCRIPTION
**Pull request by sreeramsama**
#### Description of changes
Add headless auth usage page

#### Checklist

- [x] PR description included
- [ ] `yarn test` passes
- [ ] Tests are updated
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/src/pages/[platform]/connected-components/authenticator/headless/advanced/example.react-native.mdx [hunk 1]
```diff
@@ -0,0 +1,37 @@
+```js{9,13}
+import { Amplify } from 'aws-amplify';
+import { Authenticator, useAuthenticator } from '@aws-amplify/ui-react';
```
Comment (joebuono) [inline docs/src/pages/[platform]/connected-components/authenticator/headless/advanced/example.react-native.mdx:3]: Should be `@aws-amplify/ui-react-native`?

Comment (sreeramsama) [reply docs/src/pages/[platform]/connected-components/authenticator/headless/advanced/example.react-native.mdx:3]: ah thanks for catching that

File: docs/src/pages/[platform]/connected-components/authenticator/headless/advanced/current-route.react-native.mdx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+  // Use the value of authStatus to decide which page to render
+ return (
+    <>
+      {authStatus === 'configuring' && 'Loading...'}
+      {authStatus !== 'authenticated' ? <Authenticator /> : <Home />}
+    </>
+  );
+};
+```
```
Comment (calebpollman) [inline docs/src/pages/[platform]/connected-components/authenticator/headless/advanced/current-route.react-native.mdx:53]: Let's remove this. I'm entirely sure that `authStatus` works as expected

File: docs/src/pages/[platform]/connected-components/authenticator/headless/advanced/trigger-transitions.react-native.mdx [hunk 3]
```diff
@@ -0,0 +1,11 @@
+You can use `useAuthenticator` hook to access functions that lets you trigger transitions to the authenticator. Please see [Full API](#full-api) to see all supported transition functions. Any invalid transitions (e.g. `signUp` directly to `authenticated`) will be ignored.
+
+```tsx{1,4-6}
+import { useAuthenticator } from '@aws-amplify/ui-react-native';
+
+const Home = () => {
+  const { user, signOut } = useAuthenticator((context) => [context.user]);
+
+  return <button onClick={signOut}>Welcome, {user.username}!</button>;
```
Comment (calebpollman) [inline docs/src/pages/[platform]/connected-components/authenticator/headless/advanced/trigger-transitions.react-native.mdx:9]:
```suggestion
  return <Button onPress={signOut} title={`Welcome, ${user.username}!`} />;
```

File: docs/src/pages/[platform]/connected-components/authenticator/headless/advanced/trigger-transitions.react-native.mdx [hunk 4]
```diff
@@ -0,0 +1,11 @@
+You can use `useAuthenticator` hook to access functions that lets you trigger transitions to the authenticator. Please see [Full API](#full-api) to see all supported transition functions. Any invalid transitions (e.g. `signUp` directly to `authenticated`) will be ignored.
+
+```tsx{1,4-6}
+import { useAuthenticator } from '@aws-amplify/ui-react-native';
```
Comment (calebpollman) [inline docs/src/pages/[platform]/connected-components/authenticator/headless/advanced/trigger-transitions.react-native.mdx:4]:
```suggestion
import { Button } from 'react-native';
import { useAuthenticator } from '@aws-amplify/ui-react-native';
```

File: docs/src/pages/[platform]/connected-components/authenticator/headless/advanced/useAuthenticator.react-native.mdx [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+};
+```
+
+## Authenticator Provider
+
+In advanced use cases where usage of the [`useAuthenticator` hook](headless#useauthenticator-hook) outside the scope of the [`Authenticator`](../authenticator) is needed, wrap your application inside an `Authenticator.Provider`. The `Authenticator.Provider` guarantees that the [useAuthenticator hook](headless#useauthenticator-hook) is available throughout your application. You can see an example of this pattern in the [Protected Routes Guide.](../../guides/auth-protected)
+
+```jsx
+import { Authenticator } from '@aws-amplify/ui-react-native';
+import { View } from 'react-native';
```
Comment (calebpollman) [inline docs/src/pages/[platform]/connected-components/authenticator/headless/advanced/useAuthenticator.react-native.mdx:34]: Not sure if this how all our docs are but generally think it's beet to place React and React Native imports above ours:
```suggestion
import { Text, View } from 'react-native';
import { Authenticator } from '@aws-amplify/ui-react-native';
```

File: docs/src/pages/[platform]/connected-components/authenticator/headless/advanced/useAuthenticator.react-native.mdx [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+export default function App() {
+  return (
+    <Authenticator.Provider>
+      <View>Your app here</View>
```
Comment (calebpollman) [inline docs/src/pages/[platform]/connected-components/authenticator/headless/advanced/useAuthenticator.react-native.mdx:39]: React Native can only render strings within a Text component:
```suggestion
      <View>
          <Text>Your app here</Text>
      </View>
```

File: docs/src/pages/[platform]/connected-components/authenticator/headless/advanced/current-user.react-native.mdx [hunk 7]
```diff
@@ -0,0 +1,16 @@
+You can use `useAuthenticator` hook to access current signed in `user`. If no user is authenticated, it'll return `undefined`.
+
+```tsx{1,4,8-9}
```
Comment (calebpollman) [inline docs/src/pages/[platform]/connected-components/authenticator/headless/advanced/current-user.react-native.mdx:3]: Are we sure we need the line highlights?

File: docs/src/pages/[platform]/connected-components/authenticator/headless/advanced/example.react-native.mdx [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+    <Authenticator components={components}>
+      {({ signOut, user }) => (
+        <View>
+          <Text>Hello {user.username}</Text>
+          <Button onPress={signOut}>Sign out</Button>
+        </View>
+      )}
+    </Authenticator>
+  );
+}
```
Comment (calebpollman) [inline docs/src/pages/[platform]/connected-components/authenticator/headless/advanced/example.react-native.mdx:35]: The components prop does not behave the same between React and React Native:

```suggestion
function Footer() {
  const { toResetPassword } = useAuthenticator();
  return (
    <Button onPress={toResetPassword} title="Custom Reset Password" />
  );
},

export default function App() {
  return (
    <Authenticator components={{
      SignIn: (props) => <Authenticator.SignIn {...props} Footer={Footer} />
    }}>
      <View>
        <Text>Hello {user.username}</Text>
        <Button onPress={signOut} title="Sign out" />
      </View>
    </Authenticator>
  );
}
```

File: docs/src/pages/[platform]/connected-components/authenticator/headless/advanced/useAuthenticator.react-native.mdx [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+export default () => (
+  <Authenticator.Provider>
+    <App />
+  </Authenticator.Provider>
+);
+```
+
+Then, you can use `useAuthenticator` on your App:
+
+```tsx
```
Comment (calebpollman) [inline docs/src/pages/[platform]/connected-components/authenticator/headless/advanced/useAuthenticator.react-native.mdx:19]: Can we align all the examples as being `jsx`

**Review by calebpollman — APPROVED** [review]

**Pull request merged**